jdaviz/utils.py

```diff
@@ -440,7 +441,40 @@ def get_subset_type(subset):
     if isinstance(subset.subset_state, RoiSubsetState):
         return 'spatial'
     elif isinstance(subset.subset_state, RangeSubsetState):
-        return 'spectral'
+        # look within a SubsetGroup, or a single Subset
+        subset_list = getattr(subset, 'subsets', [subset])
+
+        for ss in subset_list:
+            if hasattr(ss, 'data'):
+                ss_data = ss.data
+            elif hasattr(ss.att, 'parent'):
+                # if `ss` is a subset state, it won't have a `data` attr,
+                # check the world coordinate's parent data:
+                ss_data = ss.att.parent
+            else:
+                # if we reach this `else`, continue searching
+                # through other subsets in the group to identify the
+                # subset type:
+                continue
+
+            # check for a spectral coordinate in FITS WCS:
+            wcs_coords = (
+                ss_data.coords.wcs.ctype if hasattr(ss_data.coords, 'wcs')
+                else []
+            )
+
+            has_spectral_coords = (
+                any(str(coord).startswith('WAVE') for coord in wcs_coords) or
```

Comment:
Maybe also add an or clause for frequency? I'm not sure if the WCS has something like `FREQ` in that case.